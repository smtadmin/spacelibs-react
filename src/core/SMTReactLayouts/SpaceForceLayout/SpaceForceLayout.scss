/*
 * File: SpaceForceLayout.scss
 * Version: 0.1.0
 * Project: react-test
 * Description: Space Force Layout scss file
 * File Created: Saturday, 9th February 2021 11:01 am
 * Author: Justin Jeffrey (justin.jeffrey@siliconmtn.com)
 * -----
 * Last Modified: Wed Feb 10 2021
 * Modified By: Justin Jeffrey
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */

/** FONT IMPORTS **/

@font-face {
    font-family: "ProximaNovaMedium";
    src: local("ProximaNovaMedium"), url("./Fonts/ProximaMedium/ProximaMedium.ttf") format("truetype");
    font-weight: normal;
}

@font-face {
    font-family: "ProximaNovaRegular";
    src: local("ProximaNovaRegular"), url("./Fonts/ProximaRegular/ProximaNovaRegular.otf") format("truetype");
    font-weight: normal;
}

@font-face {
    font-family: "ProximaNovaBold";
    src: local("ProximaNovaBold"), url("./Fonts/ProximaBold/ProximaNovaBold.otf") format("truetype");
}

/** FONT PRESETS **/

h1 {
    font-family: ProximaNovaRegular;
    font-size: 48px;
}

h2 {
    font-family: ProximaNovaRegular;
    font-size: 34px;
}

h3 {
    font-family: ProximaNovaRegular;
    font-size: 24px;
}

h4 {
    font-family: ProximaNovaMedium;
    font-size: 20px;
}

p {
    font-family: ProximaNovaRegular;
    font-size: 16px;
}

body {
    font-family: ProximaNovaRegular;
    font-size: 14px;
}

.subtitle-1 {
    font-family: ProximaNovaRegular;
    font-size: 16px;
}

.subtitle-2 {
    font-family: ProximaNovaMedium;
    font-size: 14px;
}

.caption {
    font-family: ProximaNovaRegular;
    font-size: 12px;
}

.overline {
    font-family: ProximaNovaBold;
    font-size: 12px;
}

/** Colors **/

//Theme
.background-theme-light {
    background-color: #29749D;
}

.text-theme-light {
    color: #29749D;
}

.background-theme-main {
    background-color: #034E77;
}

.text-theme-main {
    color: #034E77;
}

.background-theme-dark {
    background-color: #002851;
}

.text-theme-dark {
    color: #002851;
}

//Greys
.background-main {
    background-color: #FFFFFF;
}

.background-secondary {
    background-color: #F5F5F5;
}

.background-light-grey {
    background-color: #EEEEEE;
}

.background-medium-grey {
    background-color: #B5B5B5;
}

.outline-medium-grey {
    border: solid 3px #B5B5B5;
}

.background-dark-grey {
    background-color: #616161;
}

.text-dark-grey {
    color: #616161;
}

//Message Colors
.background-success {
    background-color: #4CAF50;
}

.text-success {
    color: #4CAF50;
}

.background-error {
    background-color: #D32F2F;
}

.text-error {
    color: #D32F2F;
}

.background-info {
    background-color: #2196F3;
}

.text-info {
    color: #2196F3;
}

//Template Colors
.orange {
    background-color: #FF9800;
}

.green {
    background-color: #0A8043;
}

.light-blue {
    background-color: #039BE5;
}

.dark-blue {
    background-color: #3F51B5;
}

.purple {
    background-color: #8E24AA;
}

.brown {
    background-color: #775333;
}

.red {
    background-color: #F4511E;
}

/** THEME CSS **/@at-root

.nav-link {
    color: white !important;
}

.nav-text:hover {
    text-decoration: underline;
}

.mobile-menu {
    display: flex;
    justify-content: space-between;
}

.mobile-menu > .nav-link > .nav-icon > .nav-icon-svg {
    width: 35px !important;
    height: 35px !important;
}
