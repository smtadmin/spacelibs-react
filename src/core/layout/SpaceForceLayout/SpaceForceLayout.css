/*
 * File: SpaceForceLayout.scss
 * Version: 0.1.0
 * Project: react-test
 * Description: Space Force Layout scss file
 * File Created: Saturday, 9th February 2021 11:01 am
 * Author: Justin Jeffrey (justin.jeffrey@siliconmtn.com)
 * -----
 * Last Modified: Mon May 03 2021
 * Modified By: tylergaffaney
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */

/** FONT IMPORTS **/

@font-face {
    font-family: "NunitoSans-Regular";
    src: local("NunitoSans-Regular"),
        url("./Fonts/NunitoSans-Regular.ttf") format("truetype");
    font-weight: normal;
}

@font-face {
    font-family: "NunitoSans-Bold";
    src: local("NunitoSans-Bold"),
        url("./Fonts/NunitoSans-Bold.ttf") format("truetype");
}

@font-face {
    font-family: "NunitoSans-ExtraBold";
    src: local("NunitoSans-ExtraBold"),
        url("./Fonts/NunitoSans-ExtraBold.ttf") format("truetype");
    font-weight: normal;
}

@font-face {
    font-family: "NunitoSans-SemiBold";
    src: local("NunitoSans-SemiBold"),
        url("./Fonts/NunitoSans-SemiBold.ttf") format("truetype");
    font-weight: normal;
}

/** FONT PRESETS **/

h1 {
    font-family: NunitoSans-Regular;
    font-size: 48px;
}

h2 {
    font-family: NunitoSans-Regular;
    font-size: 34px;
}

h3 {
    font-family: NunitoSans-Regular;
    font-size: 24px;
}

h4 {
    font-family: NunitoSans-SemiBold;
    font-size: 20px;
}

h5 {
	font-family: NunitoSans-Regular;
	font-size: 1.25rem;
	display: inline-block;
}

p {
    font-family: NunitoSans-Regular;
    font-size: 16px;
}

body {
    font-family: NunitoSans-Regular;
    font-size: 14px;
}

.subtitle-1 {
    font-family: NunitoSans-Regular;
    font-size: 16px;
}

.subtitle-2 {
    font-family: NunitoSans-SemiBold;
    font-size: 14px;
}

.caption {
    font-family: NunitoSans-Regular;
    font-size: 12px;
}

.overline {
    font-family: NunitoSans-ExtraBold;
    font-size: 12px;
}

.sl-button {
    font-family: NunitoSans-ExtraBold;
}

/** Colors **/

/* Theme */
.background-theme-light {
    background-color: #29749d;
}

.text-theme-light {
    color: #29749d;
}

.background-theme-main {
    background-color: #034e77;
}

.text-theme-main {
    color: #034e77;
}

.background-theme-dark {
    background-color: #002851;
}

.text-theme-dark {
    color: #002851;
}

/* Greys */
.background-main {
    background-color: #ffffff;
}

.background-secondary {
    background-color: #f5f5f5;
}

.background-light-grey {
    background-color: #eeeeee;
}

.background-medium-grey {
    background-color: #b5b5b5;
}

.outline-medium-grey {
    border: solid 3px #b5b5b5;
}

.background-dark-grey {
    background-color: #616161;
}

.text-dark-grey {
    color: #616161;
}

/* Message Colors */

.background-success {
    background-color: #4caf50;
}

.text-success {
    color: #4caf50;
}

.background-error {
    background-color: #d32f2f;
}

.text-error {
    color: #d32f2f !important;
}

.background-info {
    background-color: #2196f3;
}

.text-info {
    color: #2196f3;
}

/* Template Colors */
.orange {
    background-color: #ff9800;
}

.green {
    background-color: #0a8043;
}

.light-blue {
    background-color: #039be5;
}

.dark-blue {
    background-color: #3f51b5;
}

.purple {
    background-color: #8e24aa;
}

.brown {
    background-color: #775333;
}

.red {
    background-color: #f4511e;
}

/** THEME CSS **/

.nav-link {
    color: white !important;
}

.nav-text:hover {
    text-decoration: underline;
}

.mobile-menu {
    display: flex;
    justify-content: space-between;
}

.mobile-menu > .nav-link > .nav-icon > .nav-icon-svg {
    width: 35px !important;
    height: 35px !important;
}

.logo {
    height: 35px;
}

.mobile-header-center-header-element-0 {
    text-align: center;
}

.question-input-wrapper {
    max-width: 300px;
}

.MuiPaper-elevation4 {
    box-shadow: none !important;
}

.MuiButton-contained {
    box-shadow: none !important;
}

.resubmit-button {
    margin-top: 20px !important;
}

.select-alt-field {
    margin-top: 5px !important;
}

.react-tabs {
	-webkit-tap-highlight-color: transparent;
	width: 100%;
}

.tab-list {
	margin: 0 0 10px;
	padding: 0;
}

.tab-list.default {
    border-bottom: 2px solid #034e77;
}

.tab-list.minimal {
    border-bottom: 2px solid #DDD;
}

.tab-list.center {
	text-align: center;
}
.tab-list.right {
	text-align: right;
}
.tab-list.left {
	text-align: left;
}

.tab.default {
    display: inline-block;
    border: 1px solid transparent;
    border-bottom: none;
    bottom: -2px;
    position: relative;
    list-style: none;
    padding: 6px 15px;
    cursor: pointer;
    border-color: #ccc;
	border-radius: 10px 10px 0 0;
	font-size: 14px;
}

.tab.minimal {
	display: inline-block;
    border-bottom: none;
    position: relative;
    list-style: none;
	border:none;
	padding: 4px 20px;
	cursor: pointer;
	font-family: NunitoSans-SemiBold;
	font-size: 14px;
	color: #034e77;
}

.tab.disabled.minimal {
	color: #aaa;
	cursor: not-allowed;
}

.tab.selected.minimal {
	border-bottom: 2px solid #034e77;
}

.tab-add {
	display: inline-block;
    border-bottom: 1px solid transparent;
    bottom: -1px;
    position: relative;
    list-style: none;
    padding: 6px 8px;
    cursor: pointer;
	border-bottom-color: #034e77;
	background-color: #DDD;
	border-radius: 10px 10px 0 0;
}

.tab.selected.default {
    background: #034e77;
    border-color: #034e77;
    color: white;
}

.tab.disabled.default {
	background: transparent;
    border-color: transparent;
	color: #aaa;
	cursor: not-allowed;
}

.react-tabs__tab--disabled {
    color: GrayText;
    cursor: default;
}

.tab:focus {
    box-shadow: 0 0 5px hsl(208, 99%, 50%);
	border-color: hsl(208, 99%, 50%);
    outline: none;
}

.react-tabs__tab:focus:after {
    content: "";
    position: absolute;
    height: 5px;
    left: -4px;
    right: -4px;
    bottom: -5px;
    background: #fff;
}

.react-tabs__tab-panel {
	display: none;
}

.react-tabs__tab-panel--selected {
    display: block;
}

div.centering {
	text-align: center;
	width:100%;
}