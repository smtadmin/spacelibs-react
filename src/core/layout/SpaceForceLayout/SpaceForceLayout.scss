/*
 * File: SpaceForceLayout.scss
 * Version: 0.1.0
 * Project: react-test
 * Description: Space Force Layout scss file
 * File Created: Saturday, 9th February 2021 11:01 am
 * Author: Justin Jeffrey (justin.jeffrey@siliconmtn.com)
 * -----
 * Last Modified: Wed Feb 17 2021
 * Modified By: Justin Jeffrey
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */

/** FONT IMPORTS **/

@font-face {
    font-family: "NunitoSans-Regular";
    src: local("NunitoSans-Regular"), url("./Fonts/NunitoSans-Regular.ttf") format("truetype");
    font-weight: normal;
}

@font-face {
    font-family: "NunitoSans-Bold";
    src: local("NunitoSans-Bold"), url("./Fonts/NunitoSans-Bold.ttf") format("truetype");
}

@font-face {
    font-family: "NunitoSans-ExtraBold";
    src: local("NunitoSans-ExtraBold"), url("./Fonts/NunitoSans-ExtraBold.ttf") format("truetype");
    font-weight: normal;
}

@font-face {
    font-family: "NunitoSans-SemiBold";
    src: local("NunitoSans-SemiBold"), url("./Fonts/NunitoSans-SemiBold.ttf") format("truetype");
    font-weight: normal;
}


/** FONT PRESETS **/

h1 {
    font-family: NunitoSans-Regular;
    font-size: 48px;
}

h2 {
    font-family: NunitoSans-Regular;
    font-size: 34px;
}

h3 {
    font-family: NunitoSans-Regular;
    font-size: 24px;
}

h4 {
    font-family: NunitoSans-SemiBold;
    font-size: 20px;
}

p {
    font-family: NunitoSans-Regular;
    font-size: 16px;
}

body {
    font-family: NunitoSans-Regular;
    font-size: 14px;
}

.subtitle-1 {
    font-family: NunitoSans-Regular;
    font-size: 16px;
}

.subtitle-2 {
    font-family: NunitoSans-SemiBold;
    font-size: 14px;
}

.caption {
    font-family: NunitoSans-Regular;
    font-size: 12px;
}

.overline {
    font-family: NunitoSans-ExtraBold;
    font-size: 12px;
}

.sl-button {
    font-family: NunitoSans-ExtraBold;
}

/** Colors **/

//Theme
.background-theme-light {
    background-color: #29749D;
}

.text-theme-light {
    color: #29749D;
}

.background-theme-main {
    background-color: #034E77;
}

.text-theme-main {
    color: #034E77;
}

.background-theme-dark {
    background-color: #002851;
}

.text-theme-dark {
    color: #002851;
}

//Greys
.background-main {
    background-color: #FFFFFF;
}

.background-secondary {
    background-color: #F5F5F5;
}

.background-light-grey {
    background-color: #EEEEEE;
}

.background-medium-grey {
    background-color: #B5B5B5;
}

.outline-medium-grey {
    border: solid 3px #B5B5B5;
}

.background-dark-grey {
    background-color: #616161;
}

.text-dark-grey {
    color: #616161;
}

//Message Colors
.background-success {
    background-color: #4CAF50;
}

.text-success {
    color: #4CAF50;
}

.background-error {
    background-color: #D32F2F;
}

.text-error {
    color: #D32F2F;
}

.background-info {
    background-color: #2196F3;
}

.text-info {
    color: #2196F3;
}

//Template Colors
.orange {
    background-color: #FF9800;
}

.green {
    background-color: #0A8043;
}

.light-blue {
    background-color: #039BE5;
}

.dark-blue {
    background-color: #3F51B5;
}

.purple {
    background-color: #8E24AA;
}

.brown {
    background-color: #775333;
}

.red {
    background-color: #F4511E;
}

/** THEME CSS **/

.nav-link {
    color: white !important;
}

.nav-text:hover {
    text-decoration: underline;
}

.mobile-menu {
    display: flex;
    justify-content: space-between;
}

.mobile-menu > .nav-link > .nav-icon > .nav-icon-svg {
    width: 35px !important;
    height: 35px !important;
}

.logo {
    height: 35px;
}
