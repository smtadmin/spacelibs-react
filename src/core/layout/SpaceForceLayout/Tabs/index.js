/*
 * File: /src/core/layout/Tabs/index.js
 * Version: 0.0.19
 * Project: @siliconmtn/spacelibs-react
 * Description: INSERT DESCRIPTION
 * File Created: Tuesday, 27th April 2021 10:05 am
 * Author: tyler Gaffaney (tyler.gaffaney@siliconmtn.com)
 * -----
 * Last Modified: Tuesday, 27th April 2021 10:05 am
 * Modified By: tyler Gaffaney (tyler.gaffaney@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */
