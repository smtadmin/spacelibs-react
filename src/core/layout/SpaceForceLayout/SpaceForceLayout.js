/*
 * File: SpaceForceLayout
 * Version: 0.1.0
 * Project: react-test
 * Description: SMT Space Force Layout
 * File Created: Wednesday, 27th January 2021 1:30 pm
 * Author: Justin Jeffrey (justin.jeffrey@siliconmtn.com)
 * -----
 * Last Modified: Tuesday, 23rd February 2021 11:00 am
 * Modified By: Justin Jeffrey (justin.jeffrey@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */

import React from 'react';
import Header from './Header/Header';
import Footer from './Footer/Footer';
import BootstrapWrapper from './BootstrapWrapper/BootstrapWrapper';
import { Container } from 'react-bootstrap';
import PropTypes from 'prop-types';
import './SpaceForceLayout.scss';
import { createMuiTheme, ThemeProvider } from '@material-ui/core/styles';

const theme = createMuiTheme({
    palette: {
        background: {
            default: "#ffffff"
        },
        primary: {
            light: '#29749D',
            main: '#034E77',
            dark: '#002851',
            contrastText: '#fff',
        },
        secondary: {
            light: '#ff7961',
            main: '#FFFFFF',
            dark: '#ba000d',
            contrastText: '#000',
        },
    },
    overrides: {
        MuiStepIcon: {
            root: {
                background: '#FFFFFF'
            },
        }
    }
});

/**
 * SMT Space Force Layout
 */
class SpaceForceLayout extends React.Component {

    /**
     * Creates an instance of the SpaceForceLayout Class
     * @param {*} props 
     */
    constructor(props) {
        super();

        /**
         * Header page locations
         *
         * @memberof SpaceForceLayout
         */
        this.header = {
            // Bootstrap for the Mobile and Desktop header content wrappers
            bootstrap: {
                mobile: "mobile-header fixed-bottom d-md-none background-theme-main",
                desktop: "desktop-header d-none d-md-flex background-theme-main white"
            },
            // Bootstrap and content to be injected in to the header left
            left: {
                mobile: {
                    content: props.ThemeConfig.header.left.mobile,
                    bootstrap: null
                },
                desktop: {
                    content: props.ThemeConfig.header.left.desktop,
                    bootstrap: null
                },
            },
            // Bootstrap and content to be injected in to the header center
            center: {
                mobile: {
                    content: props.ThemeConfig.header.center.mobile,
                    bootstrap: null
                },
                desktop: {
                    content: props.ThemeConfig.header.center.desktop,
                    bootstrap: "float-right"
                },
            },
            // Bootstrap and content to be injected in to the header right
            right: {
                mobile: {
                    content: props.ThemeConfig.header.right.mobile,
                    bootstrap: null
                },
                desktop: {
                    content: props.ThemeConfig.header.right.desktop,
                    bootstrap: null
                },
            }
        };

        /**
         * Footer page locations
         *
         * @memberof SpaceForceLayout
         */
        this.footer = {
            // Bootstrap for the Mobile and Desktop footer content wrappers
            bootstrap: {
                mobile: "mobile-footer fixed-bottom d-md-none d-flex background-theme-main justify-content-between",
                desktop: ""
            },
            // Bootstrap and content to be injected in to the footer left
            left: {
                mobile: {
                    content: props.ThemeConfig.footer.left.mobile,
                    bootstrap: null
                },
                desktop: {
                    content: props.ThemeConfig.footer.left.desktop,
                    bootstrap: null
                },
            },
            // Bootstrap and content to be injected in to the footer center
            center: {
                mobile: {
                    content: props.ThemeConfig.footer.center.mobile,
                    bootstrap: null
                },
                desktop: {
                    content: props.ThemeConfig.footer.center.desktop,
                    bootstrap: null
                },
            },
            // Bootstrap and content to be injected in to the footer right
            right: {
                mobile: {
                    content: props.ThemeConfig.footer.right.mobile,
                    bootstrap: null
                },
                desktop: {
                    content: props.ThemeConfig.footer.right.desktop,
                    bootstrap: null
                },
            }
        };


    }

    /**
     * Returns a SpaceForceLayout component
     */
    render() {

        return (
            <ThemeProvider theme={theme}>
                {/* Header Sections */}
                <Container fluid className="px-0">
                    {/* Header Desktop */}
                    <Header
                        bootstrap={this.header.bootstrap.desktop}
                        left={<BootstrapWrapper sectionName="desktop-header-left" bootstrap={this.header.left.desktop.bootstrap} content={this.header.left.desktop.content} />}
                        center={<BootstrapWrapper sectionName="desktop-header-center" bootstrap={this.header.center.desktop.bootstrap} content={this.header.center.desktop.content} />}
                        right={<BootstrapWrapper sectionName="desktop-header-right" bootstrap={this.header.right.desktop.bootstrap} content={this.header.right.desktop.content} />}
                    />
                    {/* Header Mobile */}
                    <Header
                        bootstrap={this.header.bootstrap.mobile}
                        left={<BootstrapWrapper sectionName="mobile-header-left" bootstrap={this.header.left.mobile.bootstrap} content={this.header.left.mobile.content} />}
                        center={<BootstrapWrapper sectionName="mobile-header-center" bootstrap={this.header.center.mobile.bootstrap} content={this.header.center.mobile.content} />}
                        right={<BootstrapWrapper sectionName="mobile-header-right" bootstrap={this.header.right.mobile.bootstrap} content={this.header.right.mobile.content} />}
                    />
                </Container>

                {/* This is where the content is injected */}
                {this.props.children}

                {/* Footer Sections */}
                <Container fluid>
                    {/* Desktop Footer */}
                    <Footer
                        bootstrap={this.footer.bootstrap.desktop}
                        left={<BootstrapWrapper sectionName="desktop-footer-left" bootstrap={this.footer.left.desktop.bootstrap} content={this.footer.left.desktop.content} />}
                        center={<BootstrapWrapper sectionName="desktop-footer-center" bootstrap={this.footer.center.desktop.bootstrap} content={this.footer.center.desktop.content} />}
                        right={<BootstrapWrapper sectionName="desktop-footer-right" bootstrap={this.footer.right.desktop.bootstrap} content={this.footer.right.desktop.content} />}
                    />
                    {/* Mobile Footer */}
                    <Footer
                        bootstrap={this.footer.bootstrap.mobile}
                        left={<BootstrapWrapper sectionName="mobile-footer-left" bootstrap={this.footer.left.mobile.bootstrap} content={this.footer.left.mobile.content} />}
                        center={<BootstrapWrapper sectionName="mobile-footer-center" bootstrap={this.footer.center.mobile.bootstrap} content={this.footer.center.mobile.content} />}
                        right={<BootstrapWrapper sectionName="mobile-footer-right" bootstrap={this.footer.right.mobile.bootstrap} content={this.footer.right.mobile.content} />}
                    />
                </Container>
            </ThemeProvider>
        );
    }

}

/** Prop Validators */
SpaceForceLayout.propTypes = {
    ThemeConfig: PropTypes.object,
    children: PropTypes.any
};

export default SpaceForceLayout;