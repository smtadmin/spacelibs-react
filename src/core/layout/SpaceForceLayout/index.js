/*
 * File: index
 * Version: 0.0.1
 * Project: @siliconmtn/spacelibs-react
 * Description: Index.js files are here to provide easy entry points for components. (They simplify imports.)
 * File Created: Tuesday, 16th February 2021 2:50 pm
 * Author: Justin Jeffrey (justin.jeffrey@siliconmtn.com)
 * -----
 * Last Modified: Tuesday, 16th February 2021 2:52 pm
 * Modified By: Justin Jeffrey (justin.jeffrey@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */

import SpaceForceLayout from './SpaceForceLayout';

export default SpaceForceLayout;