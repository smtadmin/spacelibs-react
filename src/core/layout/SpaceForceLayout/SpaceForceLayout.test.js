/*
 * File: SpaceForceLayout.test
 * Version: 0.1.0
 * Project: react-test
 * Description: Testing suite for the SpaceForceLayout
 * File Created: Saturday, 6th February 2021 11:01 am
 * Author: Justin Jeffrey (justin.jeffrey@siliconmtn.com)
 * -----
 * Last Modified: Wednesday, 17th February 2021 12:13 pm
 * Modified By: Justin Jeffrey (justin.jeffrey@siliconmtn.com>)
 * -----
 * Copyright 2021, Silicon Mountain Technologies, Inc.
 */
import SpaceForceLayout from './SpaceForceLayout';
import React from 'react';
import { render } from '@testing-library/react';

const ThemeConfig = {
    header: {
        left: {
            mobile: null,
            desktop: "logo"
        },
        center: {
            mobile: null,
            desktop: "main menu"
        },
        right: {
            mobile: null,
            desktop: "setting menu"
        }
    },
    footer: {
        left: {
            mobile: null,
            desktop: null
        },
        center: {
            mobile: "menu",
            desktop: null
        },
        right: {
            mobile: null,
            desktop: null
        },

    }
}

/**
 * Attempts to render a SpaceForceLayout component to confirm that it renders without crashing
 */
it("Renders without crashing", () => {

    const element = render(<SpaceForceLayout ThemeConfig={ThemeConfig} />);
    expect(element).toBeTruthy();

});

/**
 * Renders a SpaceForceLayout component and compares the render output to a snapshot to confirm that the output is consistent
 */
it("Renders Correctly", () => {

    const element = render(<SpaceForceLayout ThemeConfig={ThemeConfig} />);
    expect(element).toMatchSnapshot();
})
