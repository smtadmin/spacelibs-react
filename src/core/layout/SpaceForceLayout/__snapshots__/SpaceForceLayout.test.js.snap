// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="px-0 container-fluid"
      >
        <nav
          class="desktop-header d-none d-md-flex background-theme-main white navbar navbar-expand navbar-light"
          id="header-container"
        >
          <div
            class="header-left col-1 col"
          >
            <div
              class="desktop-header-left-direction-wrapper null row"
            >
              <div
                class="desktop-header-left-header-element-0 col"
              >
                logo
              </div>
            </div>
          </div>
          <div
            class="header-center col-10 col"
          >
            <div
              class="desktop-header-center-direction-wrapper float-right row"
            >
              <div
                class="desktop-header-center-header-element-0 col"
              >
                main menu
              </div>
            </div>
          </div>
          <div
            class="header-right col-1 col"
          >
            <div
              class="desktop-header-right-direction-wrapper null row"
            >
              <div
                class="desktop-header-right-header-element-0 col"
              >
                setting menu
              </div>
            </div>
          </div>
        </nav>
        <nav
          class="mobile-header fixed-bottom d-md-none background-theme-main navbar navbar-expand navbar-light"
          id="header-container"
        >
          <div
            class="header-left col-1 col"
          >
            <div
              class="mobile-header-left-direction-wrapper null row"
            >
              <div
                class="mobile-header-left-header-element-0 col"
              />
            </div>
          </div>
          <div
            class="header-center col-10 col"
          >
            <div
              class="mobile-header-center-direction-wrapper null row"
            >
              <div
                class="mobile-header-center-header-element-0 col"
              />
            </div>
          </div>
          <div
            class="header-right col-1 col"
          >
            <div
              class="mobile-header-right-direction-wrapper null row"
            >
              <div
                class="mobile-header-right-header-element-0 col"
              />
            </div>
          </div>
        </nav>
      </div>
      <div
        class="container-fluid"
      >
        <div
          class="row"
        >
          <div
            class="footer-left col-1 col"
          >
            <div
              class="desktop-footer-left-direction-wrapper null row"
            >
              <div
                class="desktop-footer-left-header-element-0 col"
              />
            </div>
          </div>
          <div
            class="footer-center col-10 col"
          >
            <div
              class="desktop-footer-center-direction-wrapper null row"
            >
              <div
                class="desktop-footer-center-header-element-0 col"
              />
            </div>
          </div>
          <div
            class="footer-right col-1 col"
          >
            <div
              class="desktop-footer-right-direction-wrapper null row"
            >
              <div
                class="desktop-footer-right-header-element-0 col"
              />
            </div>
          </div>
        </div>
        <div
          class="mobile-footer fixed-bottom d-md-none d-flex background-theme-main justify-content-between row"
        >
          <div
            class="footer-left col-1 col"
          >
            <div
              class="mobile-footer-left-direction-wrapper null row"
            >
              <div
                class="mobile-footer-left-header-element-0 col"
              />
            </div>
          </div>
          <div
            class="footer-center col-10 col"
          >
            <div
              class="mobile-footer-center-direction-wrapper null row"
            >
              <div
                class="mobile-footer-center-header-element-0 col"
              >
                menu
              </div>
            </div>
          </div>
          <div
            class="footer-right col-1 col"
          >
            <div
              class="mobile-footer-right-direction-wrapper null row"
            >
              <div
                class="mobile-footer-right-header-element-0 col"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="px-0 container-fluid"
    >
      <nav
        class="desktop-header d-none d-md-flex background-theme-main white navbar navbar-expand navbar-light"
        id="header-container"
      >
        <div
          class="header-left col-1 col"
        >
          <div
            class="desktop-header-left-direction-wrapper null row"
          >
            <div
              class="desktop-header-left-header-element-0 col"
            >
              logo
            </div>
          </div>
        </div>
        <div
          class="header-center col-10 col"
        >
          <div
            class="desktop-header-center-direction-wrapper float-right row"
          >
            <div
              class="desktop-header-center-header-element-0 col"
            >
              main menu
            </div>
          </div>
        </div>
        <div
          class="header-right col-1 col"
        >
          <div
            class="desktop-header-right-direction-wrapper null row"
          >
            <div
              class="desktop-header-right-header-element-0 col"
            >
              setting menu
            </div>
          </div>
        </div>
      </nav>
      <nav
        class="mobile-header fixed-bottom d-md-none background-theme-main navbar navbar-expand navbar-light"
        id="header-container"
      >
        <div
          class="header-left col-1 col"
        >
          <div
            class="mobile-header-left-direction-wrapper null row"
          >
            <div
              class="mobile-header-left-header-element-0 col"
            />
          </div>
        </div>
        <div
          class="header-center col-10 col"
        >
          <div
            class="mobile-header-center-direction-wrapper null row"
          >
            <div
              class="mobile-header-center-header-element-0 col"
            />
          </div>
        </div>
        <div
          class="header-right col-1 col"
        >
          <div
            class="mobile-header-right-direction-wrapper null row"
          >
            <div
              class="mobile-header-right-header-element-0 col"
            />
          </div>
        </div>
      </nav>
    </div>
    <div
      class="container-fluid"
    >
      <div
        class="row"
      >
        <div
          class="footer-left col-1 col"
        >
          <div
            class="desktop-footer-left-direction-wrapper null row"
          >
            <div
              class="desktop-footer-left-header-element-0 col"
            />
          </div>
        </div>
        <div
          class="footer-center col-10 col"
        >
          <div
            class="desktop-footer-center-direction-wrapper null row"
          >
            <div
              class="desktop-footer-center-header-element-0 col"
            />
          </div>
        </div>
        <div
          class="footer-right col-1 col"
        >
          <div
            class="desktop-footer-right-direction-wrapper null row"
          >
            <div
              class="desktop-footer-right-header-element-0 col"
            />
          </div>
        </div>
      </div>
      <div
        class="mobile-footer fixed-bottom d-md-none d-flex background-theme-main justify-content-between row"
      >
        <div
          class="footer-left col-1 col"
        >
          <div
            class="mobile-footer-left-direction-wrapper null row"
          >
            <div
              class="mobile-footer-left-header-element-0 col"
            />
          </div>
        </div>
        <div
          class="footer-center col-10 col"
        >
          <div
            class="mobile-footer-center-direction-wrapper null row"
          >
            <div
              class="mobile-footer-center-header-element-0 col"
            >
              menu
            </div>
          </div>
        </div>
        <div
          class="footer-right col-1 col"
        >
          <div
            class="mobile-footer-right-direction-wrapper null row"
          >
            <div
              class="mobile-footer-right-header-element-0 col"
            />
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
